wagtail/contrib/wagtailapi/serializers.py

```diff
@@ -2,171 +2,268 @@
+
+    This section is used for storing non-field data such as model name, urls, etc.
+
+    Example:
 
-    # Loop through fields
-    for field_name in fields:
-        # Check child relations
-        if field_name in child_relations and hasattr(child_relations[field_name], 'api_fields'):
-            yield field_name, [
-                dict(get_api_data(child_object, child_relations[field_name].api_fields))
-                for child_object in getattr(obj, field_name).all()
-            ]
-            continue
-
-        # Check django fields
-        try:
-            field = obj._meta.get_field(field_name)
-
-            if field.rel and isinstance(field.rel, models.ManyToOneRel):
-                # Foreign key
-                val = field._get_val_from_obj(obj)
-
-                if val:
-                    yield field_name, OrderedDict([
-                        ('id', field._get_val_from_obj(obj)),
-                        ('meta', OrderedDict([
-                             ('type', field.rel.to._meta.app_label + '.' + field.rel.to.__name__),
-                             ('detail_url', ObjectDetailURL(field.rel.to, val)),
-                        ])),
-                    ])
-                else:
-                    yield field_name, None
-            else:
-                yield field_name, field._get_val_from_obj(obj)
-
-            continue
-        except models.fields.FieldDoesNotExist:
-            pass
-
-        # Check attributes
-        if hasattr(obj, field_name):
-            value = getattr(obj, field_name)
-            yield field_name, force_text(value, strings_only=True)
-            continue
-
-
-class WagtailSerializer(BaseSerializer):
-    def to_representation(self, instance):
-        request = self.context['request']
-        fields = self.context.get('fields', frozenset())
-        all_fields = self.context.get('all_fields', False)
-        show_details = self.context.get('show_details', False)
-        return self.serialize_object(
-            request,
-            instance,
-            fields=fields,
-            all_fields=all_fields,
-            show_details=show_details
-        )
-
-    def serialize_object_metadata(self, request, obj, show_details=False):
-        """
-        This returns a JSON-serialisable dict to use for the "meta"
-        section of a particlular object.
-        """
-        data = OrderedDict()
-
-        # Add type
-        data['type'] = type(obj)._meta.app_label + '.' + type(obj).__name__
-        data['detail_url'] = ObjectDetailURL(type(obj), obj.pk)
+    "meta": {
+        "type": "wagtailimages.Image",
+        "detail_url": "http://api.example.com/v1/images/1/"
+    }
+    """
+    def get_attribute(self, instance):
+        return instance
+
+    def to_representation(self, obj):
+        return OrderedDict([
+            ('type', type(obj)._meta.app_label + '.' + type(obj).__name__),
+            ('detail_url', ObjectDetailURL(type(obj), obj.pk)),
```

Comment:
I guess the only advantage is it doesn't require the request object in the serializer context which would make unit testing and rendering APIs to static files *slightly* easier to implement (but we have the request object in the context anyway...).

So maybe you're right. It would be great to use DRFs built in renderers and that change should make the code a bit easier to follow too.